File: LLM-Translations/GPT-4/codenet/Python/Java/s048491063.java
public class s048491063 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        int M = scanner.nextInt();
        int Q = scanner.nextInt();
        int[][] train = new int[N][N];
        for (int m = 0; m < M; m++) {
            int l = scanner.nextInt();
            int r = scanner.nextInt();
            train[l - 1][r - 1]++;
        }

        int[][] accum = new int[N + 1][N + 1];
        for (int i = 0; i < N; i++) {
            int v = 0;
            for (int j = 0; j < N; j++) {
                v += train[i][j];
                accum[i + 1][j + 1] = accum[i][j + 1] + v;
            }
        }

        for (int q = 0; q < Q; q++) {
            int p = scanner.nextInt();
            int qIndex = scanner.nextInt();
            int v = accum[qIndex][qIndex] - accum[p - 1][qIndex] - accum[qIndex][p - 1] + accum[p - 1][p - 1];
            System.out.println(v);
        }
    }
}
// 

File: LLM-Translations/GPT-4/codenet/Python/C/s048491063.c
#include <stdio.h>

int main() {
    int N, M, Q;
    scanf("%d %d %d", &N, &M, &Q);
    int train[N][N];
    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            train[i][j] = 0;
        }
    }
    for (int i = 0; i < M; i++) {
        int l, r;
        scanf("%d %d", &l, &r);
        train[l - 1][r - 1]++;
    }

    int accum[N + 1][N + 1];
    accum[0][0] = 0;
    for (int i = 1; i <= N; i++) {
        accum[0][i] = 0;
        accum[i][0] = 0;
    }
    for (int i = 1; i <= N; i++) {
        for (int j = 1; j <= N; j++) {
            accum[i][j] = accum[i - 1][j] + accum[i][j - 1] - accum[i - 1][j - 1] + train[i - 1][j - 1];
        }
    }

    for (int i = 0; i < Q; i++) {
        int p, q;
        scanf("%d %d", &p, &q);
        int v = accum[q][q] - accum[p - 1][q] - accum[q][p - 1] + accum[p - 1][p - 1];
        printf("%d\n", v);
    }

    return 0;
}
// 
